discourse-doctor

```diff
@@ -0,0 +1,374 @@
+  VERIFY=`date +%s | sha256sum | base64 | head -c 20`
+  echo -e "HTTP/1.1 200 OK\n\n $VERIFY" | nc -w 4 -l -p $PORT >/dev/null 2>&1 &
+  if curl --proto =http -s $HOST:$PORT --connect-timeout 3 | grep $VERIFY
+  then
+    echo "OK"
+    return 0
+  else
+    echo "NO K"
+    return 1
+  fi
+}
+
+check_ip_match () {
+  HOST="$1"
+  echo
+  echo Checking your domain name . . . | tee -a $LOG_FILE
+  if connect_to_port $HOST 443
+  then
+      echo | tee -a $LOG_FILE
+      echo "Connection to $HOST succeeded." | tee -a $LOG_FILE
+  else
+    echo WARNING:: This server does not appear to be accessible at $HOST:443. | tee -a $LOG_FILE
+    echo | tee -a $LOG_FILE
+    if connect_to_port $HOST 80
+    then
+	echo A connection to port 80 succeeds, however. | tee -a $LOG_FILE
+	echo This suggests that your DNS settings are correct, | tee -a $LOG_FILE
+	echo but something is keeping traffic to port 443 from getting to your server. | tee -a $LOG_FILE
+	echo Check your networking configuration to see that connections to port 443 are allowed. | tee -a $LOG_FILE
+    else
+      echo "A connection to http://$HOST (port 80) also fails." | tee -a $LOG_FILE
+      echo | tee -a $LOG_FILE
+      echo This suggests that $HOST resolves to the wrong IP address | tee -a $LOG_FILE
+      echo or that traffic is not being routed to your server. | tee -a $LOG_FILE
+    fi
+    echo | tee -a $LOG_FILE
+    echo Google: \"open ports YOUR CLOUD SERVICE\" for information for resolving this problem. | tee -a $LOG_FILE
+    echo | tee -a $LOG_FILE
+    echo This test might not work for all situations, | tee -a $LOG_FILE
+    echo so if you can access Discourse at http://$HOST, this might not indicate a problem. | tee -a $LOG_FILE
+    sleep 3
+  fi
+}
+
+##
+## Do we have docker?
+##
+check_docker_version () {
+  echo  -e "\n\n==================== DOCKER INFO ====================" | tee -a $LOG_FILE
+  docker_path=`which docker.io || which docker`
+  if [ -z $docker_path ]; then
+    echo "Docker is not installed. Have you installed Discourse at all?"
+    echo "Perhaps you're looking for ./discourse-setup ."
+    echo "There is no point in continuing."
+    exit
+  else
+    echo -n "DOCKER VERSION: " | tee -a $LOG_FILE
+    docker --version | tee -a  $LOG_FILE
+    echo -e "\nDOCKER PROCESSES" | tee -a $LOG_FILE
+    sudo docker ps -a | tee -a $LOG_FILE
+  fi
+}
+
+check_OS() {
+  echo "----------------------------------------"
+  echo -n "OS:"  |tee -a $LOG_FILE
+  echo `uname -s` |tee -a  $LOG_FILE
+}
+
+check_osx_memory() {
+  echo "----------------------------------------"
+  echo -n "OSX RAM:" | tee -a $LOG_FILE
+  free -m | awk '/Mem:/ {print $2}' | tee -a  $LOG_FILE
+  free -m | tee -a $LOG_FILE
+}
+
+check_linux_memory() {
+  echo "----------------------------------------"
+  echo -n "RAM (MB):" |tee -a $LOG_FILE
+  free -m --si | awk ' /Mem:/  {print $2} ' | tee -a $LOG_FILE
+  free -m | tee -a $LOG_FILE
+}
+
+check_disk_and_memory() {
+
+  os_type=$(check_OS)
```

Comment:
so we don’t print anything if memory is low? we just print the available memory ?